hledger-lib/Hledger/Data/Journal.hs

```diff
@@ -802,7 +805,10 @@ journalConvertAmountsToCost j@Journal{jtxns=ts} = j{jtxns=map fixtransaction ts}
       fixtransaction t@Transaction{tpostings=ps} = t{tpostings=map fixposting ps}
       fixposting p@Posting{pamount=a} = p{pamount=fixmixedamount a}
       fixmixedamount (Mixed as) = Mixed $ map fixamount as
-      fixamount = canonicaliseAmount (jinferredcommodities j) . costOfAmount
+      fixamount = applyJournalStyle . costOfAmount
+      applyJournalStyle a
+          | Just s <- journalCommodityStyleLookup j (acommodity a) = a{astyle=s}
+          | otherwise = a
```

Comment:
What would it take for this to recurse forever ?
Does it ever need to look up more than twice ?

Reply:
My mistake, I thought journalCommodityStyleLookup was recursive.